### PR TITLE
Initial proposal for integer aliases

### PR DESCRIPTION
**Pull request by PiotrekDlang**
Let's start a discussion. I opt for version B.  Maybe little less convinced about uintx/intx aliases as they are quite unusual. Of course I am open for any reasonable outcome :) 
I will provide more examples and arguments later.

### CODE REVIEW

**Comment by JinShil** [comment]
Here's Rust's implementation of basically the same thing:  https://learning-rust.github.io/docs/a8.primitive_data_types.html#i8-i16-i32-i64-i128

The shorter versions are also used in the Linux Kernel (https://github.com/torvalds/linux/search?q=u8&unscoped_q=u8) but I read that was primarily due to the fact that the Linux kernel predates stdint.h.  I don't know.

I'm not too happy with the `size_t` and `ptrdiff_t` conventions borrowed from C, so I thought about potentially redefining them as `isize`, and `usize` as @PiotrekDlang  also suggested in a separate thread.  However, that convention does match the `int8`/`uint8` convention.  To match that conventions would, I guess it would be `intSize` and `uintSize`.  I think I have a slight preference to `isize` and `usize` but `intSize` and `uintSize` aren't bad.

D, however, does not over-abbreviate its names.  So, for example, Rust has `pub`, but D has `public`.  So, in a way, `int8` and friends are more idiomatic D.

Also, after having thought about it for a few days it may not be a good idea to redefine `size_t` and `ptrdiff_t`.  [They are defined in the D spec](https://dlang.org/spec/type.html#size_t), afterall.

So, what I'm thinking right now is to use `int8` and friends, and keep `size_t` and `ptrdiff_t`.  However, I would like to distinguish between types defined by the D language specification, and types introduced by this library.  So I'm thinking `size_t` and `ptrdiff_t` should go in the `d` folder, but `int8` and friends should go into a separate module, maybe `src/intTypes.d`,  Though I'd kindof like to have `float32`, `float64`, etc. to finish it off,  Perhaps those should go into `src/floatTypes.d`.

So, that's what I'm thinking right now.

File: src/d/types.d [hunk 1]
```diff
@@ -10,3 +10,53 @@ alias dstring = immutable(dchar)[];
...
+alias uint64  = ulong;
+alias uint128 = ucent;
+
+alias int8   = byte;
+alias int16  = short;
+alias int32  = int;
+alias int64  = long;
+alias int128 = cent;
+
+alias uintx = typeof(int.sizeof);
```
Comment (JinShil) [inline src/d/types.d]: Ah, didn't see these until just now.  Yeah, the `x` works.  I could go with that. 👍

**Comment by PiotrekDlang** [comment]
I'm very happy we have similar view on the issue.

Also, I just wanted to recall that the D language has deprecation process for obsolete feature. (it's a side note, as it requires a long way ;))
IMO, the initial meaning from C times for type names "size_t" (return type of sizeof operator ) and "ptrdiff_t" (return type of pointer arithmetic operation) seems to be not fully correct. Even C++ uses different definition.
That's why I started to question the "size" word in the type name. I think is very loosely related to the D definition (which I very like BTW):
"an alias to one of the unsigned integral basic types, and represents a type that is large enough to represent an offset into all addressable memory."

But I agree that "size_t" and "ptrdiff_t"  should stay as long as they are in specification.

My reasoning for the proposed B version is based on my ideal setup:

For integer types the "int8, int16, ..." scheme
For platform specific aliases:
uint - integer type for platform(memory offset) - one of integer type with specific size
int - signed version

Unfortunately "int" is a well known keyword so currently it's almost impossible to change its meaning.
So in result the "intx" version is the closest thing to the D definition, IMO.

**Comment by PiotrekDlang** [comment]
@JinShil Please take a look at the latest PR version.

- After reading your comments it I picked the longer yet more readable version of aliases again
- I had to modify the run scripts to display compiler messages in case there is error/failure

**Comment by JinShil** [comment]
Unfortunately, this isn't going to jive well with [DMD's vector extensions](https://dlang.org/spec/simd.html#x86_64_vec)

**Comment by PiotrekDlang** [comment]
Hmm. Pity.
So what now? Move to i8, i16 etc, or give up with this experiment?

**Comment by JinShil** [comment]
Even `i8`, `i16` doesn't jive with the SIMD convention.  I think the SIMD convention is a mistake.  We're lucky though.  The SIMD conventions are established in druntime itself:  https://github.com/dlang/druntime/blob/b5a95a2a85d8da441e1f2a9633c0bd0174451cb6/src/core/simd.d#L41-L66

So, we can just reimplement that our own way to use things like `int128`, `int256` instead.  So, I'm thinking we should keep what we have, and show druntime how to do it properly.

**Comment by PiotrekDlang** [comment]
OK. I will then start to write some come to test what works now. Sorry that my initial contribution is evolving so slowly. I have some limited hope that it will increase eventually ;)

**Pull request merged**